prime-router/src/main/kotlin/SubmissionReceiver.kt

```diff
@@ -268,14 +269,14 @@ class UniversalPipelineReceiver : SubmissionReceiver {
 
         when (sender.format) {
             MimeFormat.HL7 -> {
-                val messages = HL7Reader(actionLogs).getMessages(content)
-                val isBatch = HL7Reader(actionLogs).isBatch(content, messages.size)
-                // create a Report for this incoming HL7 message to use for tracking in the database
+                val messageCount = Hl7InputStreamMessageStringIterator(content.byteInputStream()).asSequence().count()
```

Comment:
I'm a little unclear on what changing a one-liner to a separate method call improves, but I'm not super familiar with Kotlin.

Reply:
I think as general practice it's good that any bit of repeated code is consolidated down to a single call. It makes it much more maintainable in the long run; if for some reason this has to be changed in the future we'll have only one place to change the code or the imports it uses if we drilled it down to a single helper function on our end rather than introducing an explicit external dependency in multiple places in our code.

Reply:
Yes, I'm aware of that. I was just under the impression that it might be different in Kotlin for some reason since that was exactly what I said about wanting to modify getMessages() instead of deleting it altogether.